packages/core/src/lib/select/searchable-select.svelte

```diff
@@ -1,241 +1,268 @@
 <!--
 @component
 
-For selecting from a list of options.
+Select an option from a list, with search
 
 ```svelte
 <SearchableSelect
   options={["Option 1", "Option 2", "Option 3"]}
-  on:input={onSelect}
+  bind:value
 />
 ```
 -->
-<svelte:options immutable />
-
 <script lang="ts">
 import cx from 'classnames';
-import { uniqueId } from '$lib/unique-id';
-import { clickOutside } from '$lib/click-outside';
-import type { IconName } from '$lib/icon';
+import { Floating } from '$lib/floating';
+import { Icon } from '$lib/icon';
 import { InputStates, type InputState } from '$lib/input';
+import { createHandleKey } from '$lib/keyboard';
+import { uniqueId } from '$lib/unique-id';
 
-import { selectControls } from './controls';
-import { createSearchableSelectDispatcher } from './dispatcher';
 import { SortOptions, getSearchResults, type SortOption } from './search';
 import SelectInput from './select-input.svelte';
-import SelectMenuButton from './select-menu-button.svelte';
-import SelectMenu from './select-menu.svelte';
 
 /** The options the user should be allowed to search and select from. */
-export let options: string[] = [];
+export let options: string[];
 
 /** The value of the search input or the currently selected option, if any. */
-export let value: string | undefined = undefined;
+export let value = '';
+
+/** The placeholder of the input. */
+export let placeholder = '';
+
+/**
+ * Whether value must be constrained to `options` on change.
+ * - `false` (Default) - any value may be selected
+ * - `true` - value must be in `options`
+ * - `Function` - value must be in `options` or pass the test function
+ */
+export let exclusive: boolean | ((value: string) => boolean) = false;
 
-/** Whether or not the select should be rendered as disabled and be non-operable. */
+/** Input is disabled. */
 export let disabled = false;
 
 /** The state of the select (info, warn, error, success), if any. */
 export let state: InputState = InputStates.NONE;
 
-/**
- * How to handle sorting for the select:
- * - `default` will sort results with a match at the beginning of a word first,
- * followed by any other results with match, followed by results with no match.
- * - 'reduce' will do the same as `default` but will filter our results with
- * no match.
- * - `off` will apply no sorting or filtering.
- */
+/** Option sorting behavior. */
 export let sort: SortOption = SortOptions.DEFAULT;
 
-/**
- * An optional call-to-action button to render at the bottom of the select menu
- * that will emit the `buttonclick` event when actioned.
- */
-export let button: { text: string; icon: IconName } | undefined = undefined;
+/** Prefix to display with the arbitrary text option, if non-exclusive. */
+export let otherOptionPrefix = '';
```

Comment:
One last addition to this PR: the app needs to be able to display a prefix on the arbitrary input for one of it's usages, e.g. `Create: some arbitrary thing`

Previously this was done by `buttontext`, but with `buttontext`, you had to thread the current value in manually. For now, all we need is a prefix, so I think the simpler, less flexible prop is easier